### PR TITLE
tests: add test_backup_volume_list to make sure that listing volume does not fail even there are bad names in backups folder

### PR DESCRIPTION
**Pull request by PhanLe1010**
test_backup_volume_list() make sure that instead of erroring the backup list should ignore the badly named files in backups folder.

longhorn/longhorn#1410

### CODE REVIEW

File: manager/integration/tests/backupstore.py [hunk 1]
```diff
@@ -213,6 +255,36 @@ def minio_write_backup_cfg_file(client, core_api, volume_name, backup_name, back
...
+    backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
+    backupstore = backup_target_setting.value
+
+    if is_backupTarget_s3(backupstore):
+        return mino_delete_file_in_backupstore(client,
+                                               core_api,
+                                               prefix,
+                                               file_name)
+
+    elif is_backupTarget_nfs(backupstore):
```
Comment (shuo-wu) [inline manager/integration/tests/backupstore.py:268]: `else:`
Why not be consistent with `backupstore_create_file`?

File: manager/integration/tests/backupstore.py [hunk 2]
```diff
@@ -170,6 +170,48 @@ def nfs_get_backup_blocks_dir():
     raise NotImplementedError
 
 
+def backupstore_create_file(client, core_api, prefix, file_name, data={}):
```
Comment (shuo-wu) [inline manager/integration/tests/backupstore.py:173]: The input args look a little bit weird at first glance. Can we use `(client, core_api, file_path, data={})` instead?
Same below.

File: manager/integration/tests/backupstore.py [hunk 3]
```diff
@@ -170,6 +170,48 @@ def nfs_get_backup_blocks_dir():
...
+
+    secret_name = backup_target_credential_setting.value
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+    bucket_name = minio_get_backupstore_bucket_name(client)
+
+    if len(data) == 0:
+        data = {"testkey": "test data from mino_create_file_in_backupstore()"}
+
+    temp_file_path = "/tmp/temp_file.cfg"
```
Comment (shuo-wu) [inline manager/integration/tests/backupstore.py:199]: It's better not to hard code `temp_file_path` in the util function.

File: manager/integration/tests/test_basic.py [hunk 4]
```diff
@@ -733,7 +729,71 @@ def test_backup_volume_list():  # NOQA
...
+                            "backup_1234@failure.cfg")
+
+    verify_no_err()
+
+    backupstore_delete_file(client,
+                            core_api,
+                            prefix,
+                            "backup_1234@failure.cfg")
+
+    bv1 = client.by_id_backupVolume(volume1_name)
```
Comment (shuo-wu) [inline manager/integration/tests/test_basic.py:788]: duplicate cleanup since the test will call `backupstore_cleanup` at the end.
Same below.

**Comment by shuo-wu** [comment]
And remember to correct the commit name.

File: manager/integration/tests/backupstore.py (hunk 1 above)
Comment (PhanLe1010) [reply manager/integration/tests/backupstore.py]: Fixed

File: manager/integration/tests/backupstore.py (hunk 2 above)
Comment (PhanLe1010) [reply manager/integration/tests/backupstore.py]: fixed

File: manager/integration/tests/backupstore.py (hunk 3 above)
Comment (PhanLe1010) [reply manager/integration/tests/backupstore.py]: fixed

File: manager/integration/tests/test_basic.py (hunk 4 above)
Comment (PhanLe1010) [reply manager/integration/tests/test_basic.py]: fixed

**Comment by PhanLe1010** [comment]
corrected the commit name

File: manager/integration/tests/common.py [hunk 5]
```diff
@@ -29,6 +29,7 @@
 EXPAND_SIZE = str(32 * Mi)
 VOLUME_NAME = "longhorn-testvol"
 DEV_PATH = "/dev/longhorn/"
+TEMP_FILE_PATH = "/tmp/temp_file.cfg"
```
Comment (shuo-wu) [inline manager/integration/tests/common.py:32]: 1. This tmp file is used in `backupstore_create_file` only, right? I think you can put this definition in the file `backupstore.py`
2. The function `backupstore_create_file` itself is a common function and it's not only for creating invalid backup config files, hence we can remove the surffix `cfg` from the tmp file name used in the function.

Comment (PhanLe1010) [reply manager/integration/tests/common.py]: Great point! Fixed. Thank you

**Pull request merged**